chirp/drivers/tdh8.py

```diff
@@ -1066,18 +1043,26 @@ def set_memory(self, mem):
             elif block_xy[1] == 8:
                 _usedflags.block8 = 0
 
-        if mem.duplex == "":
-            _mem.txfreq = mem.freq / 10
-        elif mem.duplex == "split":
-            _mem.txfreq = mem.offset / 10
-        elif mem.duplex == "+":
-            _mem.txfreq = (mem.freq + mem.offset) / 10
-        elif mem.duplex == "-":
-            _mem.txfreq = (mem.freq - mem.offset) / 10
+        if self._gmrs:
+            if 55 <= mem.number <= 199:
+                _mem.rxfreq = mem.freq / 10
+            elif 31 <= mem.number <= 55:
+                self.validate_memory(mem)
```

Comment:
Again, `validate_memory()` is called before `set_memory()` for you. You do not need to (nor should you) call it from inside `set_memory()`.